Libraries/Azure.psm1

```diff
@@ -2397,23 +2397,73 @@ function Add-AzureAccountFromSecretsFile {
 
     if ( Test-Path $secretsFile ) {
         Write-LogInfo "$secretsFile found."
-        Write-LogInfo "------------------------------------------------------------------"
-        Write-LogInfo "Authenticating Azure PS session.."
         $XmlSecrets = [xml](Get-Content $secretsFile)
         $ClientID = $XmlSecrets.secrets.SubscriptionServicePrincipalClientID
         $TenantID = $XmlSecrets.secrets.SubscriptionServicePrincipalTenantID
         $Key = $XmlSecrets.secrets.SubscriptionServicePrincipalKey
-        $pass = ConvertTo-SecureString $key -AsPlainText -Force
-        $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+        $AzureContextFilePath = $XmlSecrets.secrets.AzureContextFilePath
         $subIDSplitted = ($XmlSecrets.secrets.SubscriptionID).Split("-")
         $subIDMasked = "$($subIDSplitted[0])-xxxx-xxxx-xxxx-$($subIDSplitted[4])"
-
-        $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
-        $selectedSubscription = Select-AzureRmSubscription -SubscriptionId $XmlSecrets.secrets.SubscriptionID
+        Write-LogInfo "------------------------------------------------------------------"
+        if ($ClientID -and $Key) {
+            Write-LogInfo "Authenticating Azure PS session using Service Principal..."
+            $pass = ConvertTo-SecureString $key -AsPlainText -Force
+            $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+            $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
+        } elseif ($AzureContextFilePath) {
+            Write-LogInfo "Authenticating Azure PS session using saved context file $AzureContextFilePath..."
+            $null = Import-AzureRmContext -Path $AzureContextFilePath
+        } else {
+            $ContextFiles = Get-ChildItem -Path $PWD -Recurse | `
+                Where-Object { $_.Name.EndsWith(".json") } | Select-String -Pattern "AzureCloud" | Select Path
+            $ContextFiles = $ContextFiles.Path | Get-Unique
+            if ($ContextFiles.Count -eq 0) {
+                Write-LogWarn "No Azure authentication methods were available in Secret File."
+                Write-LogWarn "No Azure authentication context files detected in $PWD."
+
+                Write-LogWarn "Checking if current Azure powershell session is already authenticated..."
+                $selectedSubscription = Select-AzureRmSubscription -SubscriptionId $XmlSecrets.secrets.SubscriptionID -ErrorAction SilentlyContinue
+                if ( $selectedSubscription.Subscription.Id -eq $XmlSecrets.secrets.SubscriptionID ) {
+                    $SubscriptionSelected = $true
+                    Write-LogWarn "Authenticated."
+                } else {
+                    Write-LogWarn "Unable to proceed with unauthenticated Azure powershell session."
```

Comment:
revise the 2 lines log statement

Reply:
@juhlee-microsoft I'm not clear. Please explain.

Reply:
"Successfully authenticated" -> this should be info not Warn. (2428)

And "else" block should be Err not Warn. and should be terminated, right?

Reply:
@juhlee-microsoft  Fixed it.